01/pascals.md

```diff
@@ -0,0 +1,195 @@
+# Pascal's Triangle
+
+## Problem Statment
+
+Create a function, `pascals_triangle`, that returns the first `num_rows` of Pascal's Triangle. 
+
+In Pascal's triangle, each number is the sum of the two numbers directly above it as shown:
+
+![pascals triangle visualization](../images/pascal-triangle-animated.gif)
+
+The function will return a 2D array where the outer list stores the entire triangle and each index of the list stores the numbers in row `index + 1` of Pascal's triangle. For example, index 0 will store the first row, index 1 will store the second row, and so on. 
+
+The first row will always contain [1], the second row will always contain [1, 1]. Each row afterwards will have 1 as the first and last element and the middle elements will be calculated using the two numbers directly above it.
+
+If the function receives a number that is less than 0, a `ValueError` is thrown.
+
+(This problem is sourced from [LeetCode](https://leetcode.com/problems/pascals-triangle/description/))
+
```

Comment:
Do we want to specify they should use a dynamic programming approach?

Reply:
Added this specification. Thank you!